docs/arcaflow/workflows/expressions.md

```diff
@@ -28,31 +70,256 @@ $.foo.bar
+## Binary Operations
+
+Binary Operations have an expression to the left and right, with an operator in between.
+The order of operations determines which operators run first. See [Order of Operations](#order-of-operations)
+
+| Operator| Description        |
+| --------|--------------------|
+| `+`     | [Addition/Concatenation](#additionconcatenation) |
+| `-`     | [Subtraction](#subtraction) |
+| `*`     | [Multiplication](#multiplication)|
+| `/`     | [Division](#division)|
+| `%`     | [Modulus](#modulus)|
+| `^`     | [Exponentiation](#exponentiation)|
+| `==`    | [Equals](#equals)|
+| `!=`    | [Not Equals](#not-equals)|
+| `>`     | [Greater Than](#greater-than)|
+| `<`     | [Less Than](#less-than)|
+| `>=`    | [Greater Than or Equal To](#greater-than-or-equal-to)|
+| `<=`    | [Less Than or Equal To](#less-than-or-equal-to)|
+| `&&`    | [Logical And](#logical-and)|
+| `\|\|`  | [Logical Or](#logical-or)|
+
+
+### Addition/Concatenation
+
+This operator has different behavior depending on the type.
+
+##### String Concatenation
+
+When the `+` operator is used with two strings, it concatenates them together.
+For example, the expression `"a" + "b"` would output the string `"ab"`.
+
+##### Mathematical Addition
+
+When the `+` operator is used with a numerical input, it adds them together.
+For example, the expression `2 + 2` would output the integer `4`.
+
+### Subtraction
+
+When the `-` operator is used with a numerical input, it subtracts them. The operator requires numerical input.
+
+For example, the expression `6 - 4` would output the integer `2`.
+The expression `$.a + $.b` would evaluate the values of `a` and `b` within the root, and add them together.
+
+### Multiplication
+
+When the `*` operator is used with a numerical input, it multiplies them. The operator requires numerical input.
+
+For example, the expression `3 * 3` would output the integer `9`.
+
+### Division
+
+When the `/` operator is used with a numerical input, it divides them. The operator requires numerical input.
+
+For example, the expression `3 / 3` would output the integer `1`.
+
+### Modulus
+
+When the `%` operator is used with a numerical input, it outputs the remainder of the division of the operands. The operator requires numerical input.
+
+For example, the expression `2 % 3` would output the integer `2`.
+
+### Exponentiation
+
+When the `^` operator is used with numerical input, it outputs the result of the left side raised to the power of the right side.
+
+The mathematical expression 2<sup>3</sup> is represented in the expression language as `2^3`, which would output the integer `8`.
+
+### Equals
+
+The `==` equality operator checks for equality between the left and right type. It returns true when the left and right match.
+The operator currently supports the types `integer`, `float`, `string`, and `boolean`. If another type is required, please create an issue with the expected behavior of the operator with the needed type.
```

Comment:
You also can't mix types, right? So `1 == 1.0` is a syntax error, not `true`; which I think is an important point for the documentation. You need to add an explicit cast to make that work.

Reply:
Let's not talk about type casts here, since we don't have such operators, and we don't provide such functions, yet.

Reply:
I think this is the operator that will most likely have the users run into type errors. So I think it is valuable to place it here. I expanded it and referenced the functions.